src/lib/operators/operator_performance_data.hpp

```diff
@@ -21,11 +27,61 @@ struct OperatorPerformanceData : public Noncopyable {
   bool has_output{false};
   uint64_t output_row_count{0};
   uint64_t output_chunk_count{0};
+};
+
+
+/**
+ * General execution information is stored in OperatorPerformanceData through AbstractOperator::execute().
+ * The template parameter Steps allows operators to store runtimes for an operators exeuctions steps (e.g., see
+ * JoinHash). In case no steps are desired, OperatorPerformanceData should be instantiated with
+ * AbstractPerformanceData::NoSteps.
+ * Operators can further store additional execution information by inheriting from OperatorPerformanceData (e.g.,
+ * see JoinIndex).
+ */
+template <typename Steps>
+struct OperatorPerformanceData : public AbstractOperatorPerformanceData {
+  void output_to_stream(std::ostream& stream, DescriptionMode description_mode) const {
+    if (!executed) {
+      stream << "Not executed.";
+      return;
+    }
+
+    if (!has_output) {
+      stream << "Executed, but no output.";
+      return;
+    }
+
+    stream << "Output: " << output_row_count << " row" << (output_row_count > 1 ? "s" : "") << " in "
+           << output_chunk_count << " chunk" << (output_chunk_count > 1 ? "s" : "") << ", "
+           << format_duration(std::chrono::duration_cast<std::chrono::nanoseconds>(walltime)) << ".";
+
+    if constexpr (!std::is_same_v<Steps, NoSteps>) {
+      static_assert(magic_enum::enum_count<Steps>() <= sizeof(step_runtimes), "Too many steps.");
+      stream << (description_mode == DescriptionMode::SingleLine ? " " : "\n") << "Operator step runtimes:"
```

Comment:
Store `(description_mode == DescriptionMode::SingleLine ? " " : "\n")` in variable?

Reply:
I will reread the again, but I didn't like any solutions that stored separator as we have "different separators" here (it can be `, ` vs `\n` or simply ` ` vs. `\n` or `, ` vs. `\n ` for indented lists).